src/Test/TestSaveLoad.java

```diff
@@ -59,14 +62,46 @@ private boolean testSaveLoad()
     strong.initTurn(testMap);
     boolean testPassed = true;
 
-    String path = game.writeSave(false);
-    File file = new File(path);
-    testPassed &= validate(file.exists(),  "    The file we just saved to doesn't exist");
-    testPassed &= validate(GameInstance.getSaveWarnings(path).length() == 0,  "    We are incompatible with the save we just made");
-    GameInstance loaded = GameInstance.loadSave(path);
+    byte[] bytes = null;
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+         ObjectOutputStream out = new ObjectOutputStream(baos);)
+    {
+      game.writeSave(out, false);
+      bytes = baos.toByteArray();
+    }
+    catch (IOException ex)
+    {
+      System.out.println(ex.toString());
+      testPassed = false;
+    }
+    testPassed &= validate(null != bytes && bytes.length > 0,  "    Failed to generate serialized game instance.");
+
+    // test save compatibility
+    try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+         ObjectInputStream in = new ObjectInputStream(bais);)
+    {
+      testPassed &= validate(GameInstance.getSaveWarnings(in).length() == 0,  "    We are incompatible with the save we just made");
+    }
+    catch (Exception ex)
+    {
+      System.out.println(ex.toString());
+      testPassed = false;
+    }
+
+    GameInstance loaded = null;
+    try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+         ObjectInputStream in = new ObjectInputStream(bais);)
+    {
+      in.readObject(); // Pull out and discard our version info
+      loaded = (GameInstance) in.readObject();
+    }
+    catch (Exception ex)
+    {
+      System.out.println(ex.toString());
+      testPassed = false;
+    }
```

Comment:
What was the goal of the save/load changes? Whatever it was, "ease of use" was apparently not on the radar, judging by the amount of extra boilerplate here to do the same thing.
This doesn't even use the new class you added either (which would admittedly explain why this got more complicated, but doesn't explain why the new class was needed in the first place, or why it isn't being tested).

Reply:
See above.